gms/gossiper.cc

```diff
@@ -1405,6 +1405,18 @@ utils::UUID gossiper::get_host_id(inet_address endpoint) const {
     return utils::UUID(app_state->value);
 }
 
+std::set<gms::inet_address> gossiper::get_nodes_with_host_id(utils::UUID host_id) const {
+    std::set<gms::inet_address> nodes;
+    for (auto& x : get_endpoint_states()) {
+        auto node = x.first;
+        auto app_state = get_application_state_ptr(node, application_state::HOST_ID);
+        if (app_state && host_id == utils::UUID(app_state->value)) {
+            nodes.insert(node);
+        }
+    }
+    return nodes;
```

Comment:
We can not use the host_id that we use right now as the key for the map because we can have multiple entries with the same host_id, e.g., replacing a node with different ip address.

Reply:
Won't we just update the IP address? It's the same node, and the old IP address becomes useless.

Reply:
The entry of the old node is not completely useless. For instance, we need to get the tokens of the node to be replaced from the old entry.